vumi/transports/dmark/dmark_ussd.py

```diff
@@ -191,12 +198,54 @@ def handle_outbound_message(self, message):
 
         response_id = self.finish_request(
             message['in_reply_to'], json.dumps(response_data))
+
+        yield self._set_request_end(message['in_reply_to'])
+
         if response_id is not None:
-            return self.publish_ack(
+            ack = yield self.publish_ack(
                 user_message_id=message['message_id'],
                 sent_message_id=message['message_id'])
+            returnValue(ack)
         else:
-            return self.publish_nack(
+            nack = yield self.publish_nack(
                 user_message_id=message['message_id'],
                 sent_message_id=message['message_id'],
                 reason="Could not find original request.")
+            returnValue(nack)
+
+    def _set_request_start(self, message_id):
+        self.session_timestamps[message_id] = time.time()
+    
+    def _set_request_end(self, message_id):
+        start_time = self.session_timestamps.pop(message_id, None)
+        if start_time:
+            response_time = time.time() - start_time
+            if response_time > 10:
+                return self.add_status(
+                    component='response',
+                    status='down',
+                    type='slow_response',
+                    message='Very slow response',
+                    reasons=['Response took longer than 10 seconds'],
+                    details={
+                        'response_time': response_time,
+                    })
+            elif response_time > 1:
+                return self.add_status(
+                    component='response',
+                    status='degraded',
+                    type='slow_response',
+                    message='Slow response',
+                    reasons=['Response took longer than 1 second'],
+                    details={
+                        'response_time': response_time,
+                    })
+            else:
+                return self.add_status(
+                    component='response',
+                    status='ok',
+                    type='response_sent',
+                    message='Response sent',
+                    details={
+                        'response_time': response_time,
+                    })
```

Comment:
Two thoughts:
  - it would be quite useful if we could have these response times be configurable. If we could do something like `response_intervals: [10, 60]` (so the interval amounts are configurable, and the number of intervals are configurable), I think that would be ideal.
  - should some of the logic for this maybe live in the base rpc transport? Maybe we could hook into it via some callback method with a signature like `def on_response_interval(interval):` or something, and do the `self.add_status` there?

Thoughts?